front_end/content.py

```diff
@@ -1141,12 +1142,15 @@ def specify_assignment_details(self, assignment_details, introduction, date_crea
         assignment_details["has_timer"] = has_timer
         assignment_details["hour_timer"] = hour_timer
         assignment_details["minute_timer"] = minute_timer
+        assignment_details["access_restricted"] = access_restricted
```

Comment:
You might be able to convince me otherwise, but for simplicity, I think that instead of creating a new table, we should just add a single column to the assignments table called "allowed_ip_addresses". If it's empty, then there will be no restrictions based on IP address. If an instructor does want to restrict access, they can specify a comma-separated list of IP addresses that are allowed. We can store that as a single, comma-separated string in this database column. So I don't think we need to have an "access_restricted" field in this dictionary (or in the database). We can just infer that based on whether IP addresses have been separated.